internal/backend/sessions/sessionworkflows/syscalls.go

```diff
@@ -167,21 +168,38 @@ func (w *sessionWorkflow) start(ctx context.Context, args []sdktypes.Value, kwar
 }
 
 func (w *sessionWorkflow) subscribe(ctx context.Context, args []sdktypes.Value, kwargs map[string]sdktypes.Value) (sdktypes.Value, error) {
-	var (
-		connectionName string
-		filter         string
-	)
+	var name, filter string
 
-	if err := sdkmodule.UnpackArgs(args, kwargs, "connection_name", &connectionName, "filter", &filter); err != nil {
+	if err := sdkmodule.UnpackArgs(args, kwargs, "name", &name, "filter?", &filter); err != nil {
 		return sdktypes.InvalidValue, err
 	}
 
-	signalID, err := w.createEventSubscription(ctx, connectionName, filter)
+	_, connection := kittehs.FindFirst(w.data.Connections, func(c sdktypes.Connection) bool {
+		return c.Name().String() == name
+	})
+
+	_, trigger := kittehs.FindFirst(w.data.Triggers, func(t sdktypes.Trigger) bool {
+		return t.Name().String() == name
+	})
+
+	if connection.IsValid() && trigger.IsValid() {
```

Comment:
you miss the case where both are invalid

Reply:
fixed